src/deploy/osp_deployer/settings/gen_scripts/gen_prop_file.py

```diff
@@ -0,0 +1,260 @@
+                p.setKey(row["Key"])
+                role_type = p.getRoleType()
+                node_id = p.getNodeId()
+            elif row["Key"].startswith("dashboard_"):
+                node_setting_key = key.split("dashboard_")[1]
+                p = parse_key_components()
+                p.setKey(row["Key"])
+                role_type = p.getRoleType()
+                node_id = p.getNodeId()
+            elif row["Key"].startswith("controller_"):
+                p = parse_key_components()
+                p.setKey(row["Key"])
+                role_type = p.getRoleType()
+                node_id = p.getNodeId()
+                if node_id.isdigit():
+                    node_setting_key = key.split("controller_" +
+                                                 node_id + "_")[1]
+                else:
+                    continue
+            elif row["Key"].startswith("compute_"):
+                p = parse_key_components()
+                p.setKey(row["Key"])
+                role_type = p.getRoleType()
+                node_id = p.getNodeId()
+                if node_id.isdigit():
+                    node_setting_key = key.split("compute_" +
+                                                 node_id + "_")[1]
+                else:
+                    continue
+            elif row["Key"].startswith("storage_"):
+                p = parse_key_components()
+                p.setKey(row["Key"])
+                role_type = p.getRoleType()
+                node_id = p.getNodeId()
+                if node_id.isdigit():
+                    node_setting_key = key.split("storage_" +
+                                                 node_id + "_")[1]
+                else:
+                    continue
+            else:
+                continue
+
+        if role_type in settings:
+            node_settings = settings[role_type]
+        else:
+            node_settings = {}
+
+        if node_id in node_settings:
+            node_id_settings = node_settings[node_id]
+        else:
+            node_id_settings = {}
+
+        if role_type.startswith("sah"):
+            node_id_settings['is_sah'] = 'true'
+        elif role_type.startswith("director"):
+            node_id_settings['is_director'] = 'true'
+        elif role_type.startswith("dashboard"):
+            node_id_settings['is_dashboard'] = 'true'
+        elif role_type.startswith("compute"):
+            node_id_settings['is_compute'] = 'true'
+        elif role_type.startswith("controller"):
+            node_id_settings['is_controller'] = 'true'
+        elif role_type.startswith("storage"):
+            node_id_settings['is_ceph_storage'] = 'true'
+        else:
+            continue
+
+        node_id_settings[node_setting_key] = value
+        node_settings[node_id] = node_id_settings
+        settings[role_type] = node_settings
+    return settings
+
+
+def generate_tmp_file(in_data, output_file, excluded_keys, use_service_tags):
+    fData = {}
```

Comment:
This dictionary is not really needed.  Instead I would declare an empty list here.